### PR TITLE
refactor: move and add for serialized and batch items in stock summary

### PR DESCRIPTION
**Pull request by scmmishra**
## Problem
Move item feature in stock summary was broken for serialized items. 

Remove the move button

## Fix
**This PR removes the Move and Add quick entry for serialized and batch items only**. 
Clicking on the buttons will take the user to the subsequent Stock Entry form with data pre-filled. 

For other items the buttons will allow a quick entry

## Demo
Stock Summary

### CODE REVIEW

File: erpnext/stock/dashboard/item_dashboard.js [hunk 1]
```diff
@@ -16,18 +16,6 @@ erpnext.stock.ItemDashboard = Class.extend({
 		this.content = $(frappe.render_template('item_dashboard')).appendTo(this.parent);
 		this.result = this.content.find('.result');
 
-		// move
-		this.content.on('click', '.btn-move', function() {
-			erpnext.stock.move_item(unescape($(this).attr('data-item')), $(this).attr('data-warehouse'),
```
Comment (rmehta) [inline erpnext/stock/dashboard/item_dashboard.js]: create new `Stock Entry` with Item and Warehouse set (?)

**Review by rmehta — CHANGES_REQUESTED** [review]

**Comment by Anurag810** [comment]
Why we are removing this?

File: erpnext/stock/dashboard/item_dashboard.js (hunk 1 above)
Comment (scmmishra) [reply erpnext/stock/dashboard/item_dashboard.js]: Made the changes suggested.

**Comment by scmmishra** [comment]
> Why we are removing this?

Move wont work for serialized inventory. That non-standard part was not maintained along with the rest.

File: erpnext/stock/dashboard/item_dashboard.py [hunk 2]
```diff
@@ -44,7 +44,9 @@ def get_data(item_code=None, warehouse=None, item_group=None,
 
 	for item in items:
 		item.update({
-			'item_name': frappe.get_cached_value("Item", item.item_code, 'item_name')
+			'item_name': frappe.get_cached_value("Item", item.item_code, 'item_name'),
+			'disable_quick_entry': frappe.get_cached_value("Item", item.item_code, 'has_batch_no')
+									or frappe.get_cached_value("Item", item.item_code, 'has_serial_no'),
```
Comment (rohitwaghchaure) [inline erpnext/stock/dashboard/item_dashboard.py]: Indentation

File: erpnext/stock/dashboard/item_dashboard.js [hunk 3]
```diff
@@ -16,18 +16,42 @@ erpnext.stock.ItemDashboard = Class.extend({
...
-				function() { me.refresh(); });
+			let item = unescape($(this).attr('data-item'));
+			let warehouse = unescape($(this).attr('data-warehouse'));
+			let actual_qty = unescape($(this).attr('data-actual_qty'));
+			let disable_quick_entry = Number(unescape($(this).attr('data-disable_quick_entry')));
+			let rate = unescape($(this).attr('data-rate'));
+
+			if (disable_quick_entry) open_stock_entry(item, warehouse);
+
+			else erpnext.stock.move_item(item, null, warehouse, actual_qty, rate, function() { me.refresh(); })
```
Comment (rohitwaghchaure) [inline erpnext/stock/dashboard/item_dashboard.js]:
```
if () {

} else {

}
```
easy to read than 
```
if ()
else 
```

File: erpnext/stock/dashboard/item_dashboard.js [hunk 4]
```diff
@@ -16,18 +16,47 @@ erpnext.stock.ItemDashboard = Class.extend({
...
-			erpnext.stock.move_item(unescape($(this).attr('data-item')), $(this).attr('data-warehouse'),
-				null, $(this).attr('data-actual_qty'), null, function() { me.refresh(); });
+			handle_move_add($(this), "Move")
 		});
 
 		this.content.on('click', '.btn-add', function() {
-			erpnext.stock.move_item(unescape($(this).attr('data-item')), null, $(this).attr('data-warehouse'),
-				$(this).attr('data-actual_qty'), $(this).attr('data-rate'),
-				function() { me.refresh(); });
+			handle_move_add($(this), "Move")
```
Comment (rohitwaghchaure) [inline erpnext/stock/dashboard/item_dashboard.js]: Should be "Add" instead of Move?

**Review by rohitwaghchaure — APPROVED** [review]

**Pull request merged**